docs/sources/source/tutorials/old_callbacks.rst

```diff
@@ -0,0 +1,206 @@
+:orphan:
+
+.. _callbacks0.10:
+
+Callbacks V0.10
+===============
+NeMo uses callbacks to do a variety of helper functions during training.
+NeMo comes with three useful callbacks: SimpleLossLoggerCallback,
+CheckpointCallback, and EvaluatorCallback. Callbacks are defined prior to
+calling the train() function, and are passed to the train() function.
+For example, a common training script will look like:
+
+.. code-block:: python
+
+    loggercallback = nemo.core.SimpleLossLoggerCallback(...)
+    savercallback = nemo.core.CheckpointCallback(...)
+    evalcallback = nemo.core.EvaluatorCallback(...)
+
+    nf.train(
+        callbacks=[loggercallback, savercallback, evalcallback],
+        ...)
+
+SimpleLossLoggerCallback
+------------------------
+SimpleLossLoggerCallback is used to
+log training metrics such as loss and time per step to screen and to
+tensorboard. SimpleLossLoggerCallback has one required arguments, and two
+arguments that we recommend to overwrite. It requires tensors which is a list
+of NMTensors that print_func(), get_tb_values(), and log_to_tb_func will
+receive as input during
+training. The two reccomended arguments to override are print_func(), and
+either get_tb_values() or log_to_tb_func().
+
+print_func() should be used to log values to screen. We recommend using
+logging.info() in place
+of print(). For example, it can be used to print the loss value:
+
+.. code-block:: python
+
+    from nemo import logging
+    def my_print_func(tensors):
+        logging.info(f"Loss {tensors[0]}")
+
+We provide two methods to log to tensorboard: get_tb_values() and
+log_to_tb_func(). For simple use case of logging scalars, we recommend
+get_tb_values(). For advanced use cases such as pictures or audio, we
+recommend log_to_tb_func().
+
+get_tb_values() is used to return values to be logged to tensorboard. It should
+return a list of 2-element tuples, where the first element is a string
+representing the tensorboard tag, and the second element is the scalar value to
+log. Note we currently only support scalar values. Note to use get_tb_values(),
+tb_writer should also be defined.
+
+.. code-block:: python
+
+    def my_get_tb_values(tensors):
+        return [("Train_Loss", tensors[0])]
+
+log_to_tb_func() takes three arguments: the
+`tensorboardX.SummaryWriter <https://tensorboardx.readthedocs.io/en/latest/tensorboard.html>`_
+, a list of evaluated tensors, and the current step. The user can then use the
+SummaryWriter class to add images, audio, and more. For example:
+
+.. code-block:: python
+
+    def log_to_tb_func(swriter, tensors, step):
+        swriter.add_scalar("Train_Loss", tensors[0], step)
+        swriter.add_audio("Train_Sample", tensors[1][0], step)
+
+SimpleLossLoggerCallback can be constructed as follows:
+
+.. code-block:: python
+
+    from functools import partial
```

Comment:
Do we need this import here?

Reply:
Are these changes (2/3) important? Note that this is old_callbacks.rst, which covers the deprecated callbacks and are planned for removal in v0.12.

I think partial is a cool function that most people don't know exists (incl. me before I found out) and it is used in this codeblock.

Reply:
Yap, a cool function, but I cannot see _any reference_ to this function...

Reply:
it has been removed